lib/ansible/modules/cloud/amazon/aws_kms.py

```diff
@@ -170,7 +170,7 @@ def do_grant(kms, keyarn, role_arn, granttypes, mode='grant', dry_run=True, clea
                     if clean_invalid_entries and invalid_entries:
                         # we have bad/invalid entries. These are roles that were deleted.
                         # prune the list.
-                        valid_entries = filter(lambda x: x.startswith('arn:aws:iam::'), statement['Principal']['AWS'])
+                        valid_entries = list(filter(lambda x: x.startswith('arn:aws:iam::'), statement['Principal']['AWS']))
```

Comment:
I'd rather get rid of the `filter` and `lambda` entirely:

```
valid_entries = [item for item in statement['Principal']['AWS'] if not item.startswith('arn:aws:iam::')]
```

Reply:
'k. I changed it in both places, please sanity check my boolean logic.